e2e_testing/torchscript/elementwise.py

```diff
@@ -339,10 +339,65 @@ def forward(self, x):
         return torch.gt(x, 0.6)
 
 
-@register_test_case(module_factory=lambda: ElementwiseGtScalarModule())
-def ElementwiseGtScalarModule_basic(module, tu: TestUtils):
+@register_test_case(module_factory=lambda: ElementwiseGtFloatScalarModule())
+def ElementwiseGtFloatScalarModule_basic(module, tu: TestUtils):
     module.forward(tu.rand(3, 5))
 
+
+class ElementwiseGtIntScalarModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.int64, True),
+    ])
+    def forward(self, x):
+        return torch.gt(x, 10)
+
+
+@register_test_case(module_factory=lambda: ElementwiseGtIntScalarModule())
+def ElementwiseGtIntScalarModule_basic(module, tu: TestUtils):
+    module.forward(torch.randint(-10, 15, (3,4)))
+
+
+class ElementwiseGtMixedScalarModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.int64, True),
+    ])
+    def forward(self, x):
+        return torch.gt(x, 7.0)
```

Comment:
Here the important test is making sure that we are not rounding the float value to an integer. So you want to make sure that `torch.gt(x, 7.7)` and `torch.gt(x, 7.3)` work correctly. (they currently will not work correctly because the lowering is turning the float into an int).

It's okay to have features not supported, as long as they result in an unimplemented error. Since the functionality you want right now is `torch.gt(int, int)`, I would just make sure in the `int` branch of the lowering that both the scalar and the tensor are integers. Then leave this case as unimplemented and for a future patch to solve.